LFPy/test/test_recextelectrode.py

```diff
@@ -177,8 +177,125 @@ def test_method_soma_as_point_contact_average_r10n100(self):
+            stick.set_pos(z=-stick.zstart[0])
+
+            synapse = LFPy.StimIntElectrode(stick, stick.get_closest_idx(0, 0, 1000),
+                                   **stimParams)
+            stick.simulate([isotropic_electrode, anisotropic_electrode],
+                           rec_imem=True, rec_istim=True, rec_vmem=True)
+
+            np.testing.assert_allclose(isotropic_electrode.LFP,
+                                       anisotropic_electrode.LFP)
+
+    def test_compare_anisotropic_lfp_methods(self):
+
+        stickParams = {
+            'morphology' : os.path.join(LFPy.__path__[0], 'test', 'stick.hoc'),
+            'passive_parameters' : {'g_pas' : 1./30000, 'e_pas' : -65},
+            'passive': True,
+            'tstart' : 0,
+            'tstop' : 20,
+            'dt' : 2**-4,
+            'nsegs_method' : 'lambda_f',
+            'lambda_f' : 1000,
+
+        }
+        stimParams = {
+            'pptype' : 'SinSyn',
+            'delay' : -100.,
+            'dur' : 1000.,
+            'pkamp' : 1.,
+            'freq' : 100.,
+            'phase' : -np.pi/2,
+            'bias' : 0.,
+            'record_current' : True
+        }
+
+        electrodeParams = {
+            'sigma' : [0.3, 0.3, 0.45],
+            'x' : np.array([0, 1000]),
+            'y' : np.zeros(2),
+            'z' : np.zeros(2),
+
+        }
+
+        ps_electrodeParams = electrodeParams.copy()
+        ls_electrodeParams = electrodeParams.copy()
+        sap_electrodeParams = electrodeParams.copy()
+
+        ps_electrodeParams["method"] = "pointsource"
+        ls_electrodeParams["method"] = "linesource"
+        sap_electrodeParams["method"] = "soma_as_point"
+
+        electrode_ps = LFPy.RecExtElectrode(**ps_electrodeParams)
+        electrode_ls = LFPy.RecExtElectrode(**ls_electrodeParams)
+        electrode_sap = LFPy.RecExtElectrode(**sap_electrodeParams)
+
+        stick = LFPy.Cell(**stickParams)
+        stick.set_pos(z=-stick.zstart[0])
+
+        synapse = LFPy.StimIntElectrode(stick, stick.get_closest_idx(0, 0, 1000),
+                               **stimParams)
+        stick.simulate([electrode_ps, electrode_ls, electrode_sap],
+                       rec_imem=True, rec_istim=True, rec_vmem=True)
+
+        # Test that distant electrode is independent of choice of method
+        np.testing.assert_almost_equal(electrode_ps.LFP[1,:],
+                                   electrode_ls.LFP[1,:])
+
+        np.testing.assert_almost_equal(electrode_ps.LFP[1,:],
+                                   electrode_sap.LFP[1,:])
+
+        # Hack to test that LFP close to stick is dependent on choice of method
+        np.testing.assert_raises(AssertionError, np.testing.assert_array_equal,
+                                 electrode_ps.LFP[0,:], electrode_ls.LFP[0,:])
+
+        np.testing.assert_raises(AssertionError, np.testing.assert_array_equal,
+                                 electrode_ps.LFP[0,:], electrode_sap.LFP[0,:])
```

Comment:
I wonder if we should also create a separate test set/test file for the lfpcalc module in LFPy. 
It is now rather redundant to include this analytical LFP in each test. It should be quite straightforward to implement tests for each public method implementation like `LFPy/test/test_lfpcalc.py`:
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""Copyright (C) 2012 Computational Neuroscience Group, NMBU.

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

"""
from __future__ import division
import os
import unittest
import numpy as np
import LFPy

class testLfpCalc(unittest.TestCase):
    """
    test class LFPy.lfpcalc
    """
    def test_calc_lfp_pointsource(self):
        sigma = 0.3
        cell = TestCell()
        np.testing.assert_equal(1./(4*np.pi*sigma),
                                LFPy.lfpcalc.calc_lfp_pointsource(cell,
                                    x=1, y=0, z=0, sigma=sigma,
                                    r_limit=cell.diam/2))

class TestCell(object):
    """Cell like object with attributes for predicting extracellular potentials,
    but with:
        - 1 compartment
        - position in (0,0,0)
        - length 1
        - diam 1
        - current amplitude 1
        - 1 timestep
    """
    def __init__(self):
        self.imem = np.array([[1.]])
        self.xstart = np.array([0.])
        self.ystart = np.array([0.])
        self.zstart = np.array([0.])
        self.xmid = np.array([0.])
        self.ymid = np.array([0.])
        self.zmid = np.array([0.])
        self.xend = np.array([0.])
        self.yend = np.array([0.])
        self.zend = np.array([0.])
        self.diam = np.array([1.])
    

```